test/greenlight/test_suite/yellow.clj

```diff
@@ -0,0 +1,45 @@
+(ns greenlight.test-suite.yellow
+  (:require
+    [clojure.test :refer [is]]
+    [com.stuartsierra.component :as component]
+    [greenlight.step :as step :refer [defstep]]
+    [greenlight.test :as test :refer [deftest]]))
+
+
+(def counter (atom 0))
+
+
+(defstep reset-counter
+  "Reset the counter."
+  :title "Reset counter"
+  :test (fn [_]
+          (reset! counter 0)))
+
+
+(defstep fail-until-3rd-try
+  "A step that fails until the 3rd try."
+  :title "Fail Until 3rd Try"
+  :test (fn [_]
+          (swap! counter inc)
+          (is (= @counter 3))))
```

Comment:
Note: `swap!` returns the updated value. It's not really an issue here, but in general this is not a safe pattern for interacting with state - concurrent access to the atom could have changed the value between these two lines, and then your test may never evaluate to true.
```suggestion
          (is (= 3 (swap! counter inc)))))
```
I've also inverted the ordering, because `clojure.test/is` expects the _first_ argument to `=` to be the expected value, and the second to be the actual value. Otherwise you'll get confusing error messages when this fails, like "Expected to see 2 but got 3".